docs: refactor "Restore and Backup". Add details on creating an IAM role for cluster backup (#1460)
spectrocloud/librarium: content/docs/04-clusters/06-cluster-management/05-backup-restore.md

import InfoBox from 'shared/components/InfoBox';
import PointsOfInterest from 'shared/components/common/PointOfInterest';

# Cluster Backup and Restore

> A backup is a persistent state of Kubernetes clusters, including the native objects such as Pods, DaemonSets, Services, and persistent volumes. A backup allows you to save the current state of a cluster and restore it at a later point in time if needed. You can restore a backup to the same or a different cluster. Internally, Palette leverages an open-source tool called *Velero* to provide backup and restore capabilities. 

Reviewer: ```suggestion
A backup is a persistent state of Kubernetes resources, ranging from objects such as Pods, DaemonSets, Services, to persistent volumes. A backup allows you to save the current state of a cluster and restore it at a later point in time if needed. You can restore a backup to the same or a different cluster. 
```
The first sentence had improve use of plurals - "the native objects", "the" is incorrect in this usage. It would work if you remove the article, so it stated "including native objects" but the article makes it specific and that's where the plural crashes.

A backup can be specific to a set of resources, not necessarily the cluster itself, that's why the sentence is changed in the suggestion.

> Palette allows you to schedule and take a backup of a specific cluster or an entire [workspace](/workspace). Also, you can maintain multiple backups of a cluster or a workspace. This guide provides the instructions for creating and restoring a cluster backup. You can refer to the [Manage Palette Workspace](/workspace/workload-features#managepaletteworkspace) guide to create and restore a workspace backup. 

Reviewer: ```suggestion
You can schedule and take a backup of a specific cluster or an entire [workspace](/workspace). Also, you can maintain multiple backups of a cluster or a workspace. This guide provides instructions for creating and restoring a cluster backup. You can refer to the [Manage Palette Workspace](/workspace/workload-features#managepaletteworkspace) guide to learn more about backup and restore actions for a workspace. 
```
Use an active voice. By removing Palette, you can address the user directly in the intro sentence.

> Before you create a backup, the initial step is to configure a backup location in Palette. A backup location is an object storage in the cloud. Palette supports the following object storages as the backup location.

Reviewer: ```suggestion
Before you create a backup, the initial step is to configure a backup location in Palette. A backup location is an object storage. Palette supports the following object storages as the backup location.
```
It doesn't have to be in the cloud, you can deploy MinIO to an on-prem data center. Be careful about assuming something is only for the cloud. We deal with both on-prem and cloud, so we have to be careful when stating what environment something is available in.

- Amazon Web Services (AWS) Simple Storage Service (S3) bucket

- Google Cloud Platform (GCP) bucket

- MinIO S3 bucket

- Azure blob storage

> While configuring a backup location, you authorize Palette to perform S3 operations in the cloud account. An outstanding use case is when the backup cloud account differs from the one where Palette deploys the clusters. In such scenarios, you must also authorize the latter cloud account, as the diagram below highlights. The numerical points 1, 2, and 3 highlighted in the diagram below signify the following sequence you follow in Palette.

Reviewer: ```suggestion
When you create a backup location, you authorize Palette to perform S3 operations in the environment the object storage resides. A common use case is to back up a cloud account that is different from the one where Palette deploys clusters. In this scenario, you must authorize the latter cloud account, as the diagram below highlights. The numerical points 1, 2, and 3 highlighted in the diagram below signify the following sequence you follow in Palette.
```
Use active voice in the writing. It's also a bit strange how this transitions from an overview into an advanced scenario. 

This advanced usecase should be explained separately and not in the overview. The advanced use case should have an example that makes this clearer. The diagram is great, and helps the reader better understand the concept, but we should also have a dedicated example.

<br />

1. When registering a primary cloud account with Palette, you authorize Palette to deploy clusters in the cloud account. 

2. When adding a secondary cloud account as the backup location, you authorize Palette to store the backup in the cloud account. The secondary cloud account can be the same or different than the primary cloud account. 

3. If the secondary cloud account differs from the primary one, you must update the blob storage's access management rules to allow the primary cloud account access to the storage.

![A diagram highlighting a use-case when the backup cloud account is different from the cluster deployment cloud account.](/clusters_cluster-management_backup-restore_separate-cloud-accounts.png)

You can establish the authorization or a trust relationship between the cloud entities per the guidelines of your cloud service provider. For example, if you configuring the backup in AWS, you can use Security Token Service (STS), IAM roles, and a trust relationship. In the case of Azure, you can use Shared Access Signatures (SAS), AD roles, and service principal identities. Similarly, you can use service accounts, credentials, and IAM roles for GCP. 

> 

Reviewer: We need a title heading here. This is where you start explaining how to create a backup location. Right now there is no transition between the intro and this.
